java-benchmarks/mx.java-benchmarks/mx_java_benchmarks.py

```diff
@@ -1924,6 +1924,14 @@ def postprocessRunArgs(self, benchname, runArgs):
             else:
                 return ["-r", str(iterations)] + remaining
 
+    def vmArgs(self, bmSuiteArgs):
+        vm_args = super(RenaissanceBenchmarkSuite, self).vmArgs(bmSuiteArgs)
+        if java_home_jdk().javaCompliance > '16':
+            vm_args += ["--add-opens", "java.management/sun.management=ALL-UNNAMED"]
```

Comment:
@farquet should this be done in the Renaissance suite itself?

Reply:
Yes, Renaissance has some `Add-Opens` entries directly in the Manifest: https://github.com/renaissance-benchmarks/renaissance/pull/244/files

But why is it needed for native image but not for the JIT?

Reply:
@farquet We need these exports for both NI and JVM. The only difference is that in the case of NI, we are automatically doing export.

Reply:
> [2021-11-11T08:16:00.039+0100] com.twitter.jvm.Jvm$ (com.twitter.jvm.Jvm$ liftedTree2$1)
> WARNING: failed to create Hotspot JVM interface, using NilJvm instead
> [2021-11-11T08:16:00.397+0100] com.twitter.finagle (com.twitter.finagle.Init$ $anonfun$once$1)
> INFO: Finagle version 19.4.0 (rev=15ae0aba979a2c11ed4a71774b2e995f5df918b4) built at 20190418-114348
> Master port: 40705
> Cache ports: 44385, 41639, 35635, 43851, 39261, 40521, 36315, 36317, 37231, 46291, 33635, 33083
> ====== finagle-chirper (web) [default], iteration 0 started ======
> Resetting master, feed map size: 5000
> GC before operation: completed in 25.968 ms, heap usage 113.254 MB -> 27.470 MB.
> WARNING: This benchmark provides no result that can be validated.
> There is no way to check that no silent failure occurred.
> ====== finagle-chirper (web) [default], iteration 0 completed (21390.592 ms) ======

This is the output of the JVM execution. As you can see, the problem is that the JVM fails to create an HS JVM interface and fallbacks to the NilJVM interface (because of missing exports). I think that this is not intended to happen.

Reply:
Ah ok, got it. This is a good catch. We should then add the add-opens upstream just like in the PR I've shared above. Would you be ok to open a PR to renaissance directly?

On the graal mx side, adding it on the command line like you did makes sense then, but I would guard it to be added only for the renaissance versions up to 0.13.0 (since we'll add the export upstream for the upcoming releases)

Reply:
Sure, here is the [PR](https://github.com/renaissance-benchmarks/renaissance/pull/330). I will also add a check if the renaissance version is 0.13.0. :smile:

Reply:
Thanks!
Please make sure the test for version also includes versions earlier than 0.13.0.

Reply:
Sure! I have added checks for earlier versions as well.